tests/test_dynamic_environments.py

```diff
@@ -174,44 +206,131 @@ def test_merge(self):
             "C": "D"
         }
 
-        data = acre.merge(data, current_env=environment)
-
-        # Note that C from the environment is not preserved, it's overwritten
-        # by the merge. (To preserve it should have a reference to itself)
-        # See "A"
-        self.assertEqual(data, {
-            "A": "a;A",
-            "B": "b;c;d",
-            "C": "C"
-        })
+        # test Windows platform
+        with mock.patch("platform.system", return_value="Windows"):
+            with mock.patch("os.pathsep", ";"):
+                data = acre.merge(data, current_env=environment)
+
+                # Note that C from the environment is not preserved,
+                # it's overwritten by the merge. (To preserve it should have
+                # a reference to itself). See "A"
+                self.assertEqual(data, {
+                    "A": "a;A",
+                    "B": "b;c;d",
+                    "C": "C"
+                })
+
+        # test unix platforms
+        with mock.patch("platform.system", return_value="Linux"):
+            with mock.patch("os.pathsep", ":"):
+                data = {
+                    "A": "a:{A}",
+                    "B": "b:c:d",
+                    "C": "C"
+                }
+                data = acre.merge(data, current_env=environment)
+                self.assertEqual(data, {
+                    "A": "a:A",
+                    "B": "b:c:d",
+                    "C": "C"
+                })
```

Comment:
See my big comment above. Basically the idea is that even for Mac OS / Linux that the "input data" is written with `;` as separator and only when it is about to be applied should it become the separator for the specific OS, basically saying "use this separator for the final paths".

That means that technically we shouldn't be mocking a different "data" input set for Linux/OS X yet only the output set should result in different data. I'm not sure my original intentions and its logic is sound, but that was definitely the idea. One system (one separator) that outputs final data into the format needed for the specific OS.